js/src/index.ts

```diff
@@ -229,8 +232,14 @@ export class Schematic {
   };
 
   // Send track event
-  track = (body: EventBodyTrack): void => {
-    this.handleEvent("track", body);
+  track = (body: EventBodyTrack): Promise<void> => {
```

Comment:
I changed these track and identify calls to return Promise<void> instead of void, so that I could wait for them in tests. I think this should be fine as callers can still just choose not to await them.